parsec/api/protocol/handshake.py

```diff
@@ -331,12 +353,16 @@ class BaseClientHandshake:
     SUPPORTED_API_VERSIONS: Sequence[ApiVersion]  # Overwritten by subclasses
 
     def __init__(self) -> None:
-        self.challenge_data: Dict[str, object]
+        self.challenge_data: Dict[str, object] = {"client_timestamp": self.timestamp()}
```

Comment:
Not sure about this since the challenge is supposed to be data sent by the server...
Why not having `client_timestamp` as a proper field next to `client_api_version` ?

Reply:
The reason why `client_timestamp` is added to the challenge data is because I wanted this dict to have the same fields as:
https://github.com/Scille/parsec-cloud/blob/8361fb3fb81dedd150829313b334a7f21b7cdab4/parsec/api/protocol/base.py#L82-L89
where `client_timestamp` is returned by the backend. In the case of the handshake, the client timestamp cannot be sent to the backend and incorporated in its challenge request because there no client request before that, the client simply connects and gets a challenge request, which means the client timestamp has to be generated before connecting.

I can add a comment if that's ok.

Reply:
I moved the logic a bit to make it more obvious: 36a69c35bb4aad418c9a77d3f25478cb5e0049a8